mods/sib-plusplus-tweaker.wh.cpp

```diff
@@ -0,0 +1,453 @@
+
+    DWORD old;
+    if(mod_settings.DHTracking == TRUE)
+    {
+        Wh_Log(L"- Disable Hottracking...");
+        uint8_t* p_trackingBG = (uint8_t*)(FindPattern("48 8B C4 48 89 50 10 48 89 48 08",(uintptr_t)g_hStartIsBackModule));
+        uint8_t* p_trackingBorder = (uint8_t*)(FindPattern("4C 8B DC 49 89 6B 18 49 89 73 20",(uintptr_t)g_hStartIsBackModule));
+
+        VirtualProtect(p_trackingBG, 1, PAGE_EXECUTE_READWRITE, &old);
+        p_trackingBG[0] = 0xC3;
+        VirtualProtect(p_trackingBG, 1, old, 0);
+        
+        VirtualProtect(p_trackingBorder, 1, PAGE_EXECUTE_READWRITE, &old);
+        p_trackingBorder[0] = 0xC3;
+        VirtualProtect(p_trackingBorder, 1, old, 0);
+    }
+    if(mod_settings.DisableCustomOrb == TRUE)
+    {
+        Wh_Log(L"- Disable Custom Orb...");
+        uint8_t* p_createOrb = (uint8_t*)(FindPattern("40 56 48 83 EC 40 48 8D 15 03 70 01 00",(uintptr_t)g_hStartIsBackModule));
+        uint8_t* p_winKeyHook = (uint8_t*)(FindPattern("40 53 48 81 EC A0 00 00 00",(uintptr_t)g_hStartIsBackModule));
+
+        VirtualProtect(p_createOrb, 1, PAGE_EXECUTE_READWRITE, &old);
+        p_createOrb[0] = 0xC3;
+        VirtualProtect(p_createOrb, 1, old, 0);
+        
+        VirtualProtect(p_winKeyHook, 1, PAGE_EXECUTE_READWRITE, &old);
+        p_winKeyHook[0] = 0xC3;
+        VirtualProtect(p_winKeyHook, 1, old, 0);
+    }
+    if(mod_settings.DisableCDSB == TRUE)
+    {
+        Wh_Log(L"- Disable Custom Scrollbar...");
+        uint8_t* p_cdScrollbar = (uint8_t*)(FindPattern("40 55 41 56 48 83 EC 28 4C 8B F1",(uintptr_t)g_hStartIsBackModule));
+
+        VirtualProtect(p_cdScrollbar, 2, PAGE_EXECUTE_READWRITE, &old);
+        p_cdScrollbar[0] = 0xC3;
+        p_cdScrollbar[1] = 0xC3;
+        VirtualProtect(p_cdScrollbar, 2, old, 0);
+    }
+    if(mod_settings.DisableSMAnim == TRUE)
+    {
+        Wh_Log(L"- Disable Start Menu Animations...");
+        uint8_t* p_smAnimations = (uint8_t*)(FindPattern("41 8D 51 08 8D 4A 40 FF 15 CA 7B 05 00",(uintptr_t)g_hStartIsBackModule));
+
+        VirtualProtect(p_smAnimations, 7, PAGE_EXECUTE_READWRITE, &old);
+        for(int i=0; i < 7; i++) p_smAnimations[i] = 0x90;
+        VirtualProtect(p_smAnimations, 7, old, 0);
+    }
+    if(mod_settings.ForceDWM == TRUE)
+    {
+        Wh_Log(L"- Force DWM Composition (Taskbar & Start Menu)...");
+        uint8_t* p_dwmComposition = (uint8_t*)(FindPattern("48 8B CB E8 43 1D 03 00 84 C0 75 4E",(uintptr_t)g_hStartIsBackModule)) - 9;
+
+        VirtualProtect(p_dwmComposition, 9, PAGE_EXECUTE_READWRITE, &old);
+        for(int i=0; i < 9; i++) p_dwmComposition[i] = 0x90;
+        VirtualProtect(p_dwmComposition, 9, old, 0);
+    }
+    if(mod_settings.MatchSevenFolders == TRUE)
+    {
+        Wh_Log(L"- Match 7 Start Menu Links...");
+        SpecialFoldersList = decltype(SpecialFoldersList)(REL((uintptr_t)(FindPattern("48 8D 35 ?? ?? ?? ?? 4C 8B F1 4C 8D 2D ?? ?? ?? ?? 33 FF",(uintptr_t)g_hStartIsBackModule)),3) - 8);
```

Comment:
You're passing it to `REL(` which evaluates it twice, so a pattern will be searched twice. For reasons such as this, I'd avoid using macros like these unless absolutely necessary.

Reply:
It kind of is necessary. Improved it a little.

Reply:
I mean, it's unnecessary for it to be a macro. It can be a function, can't it?

Reply:
It sure can. I find macros easy to use if done properly. I should probably note that this specific snippet was made by someone else who I credited and got permission from.